client/lib/safe-image-url/index.js

```diff
@@ -80,7 +80,9 @@ export default function safeImageUrl( url ) {
 		delete parsedUrl.search;
 
 		// Force https since if there's a missing protocol it'll throw an error
-		parsedUrl.protocol = 'https';
+		if ( ! parsedUrl?.protocol ) {
```

Comment:
No need for `?.` here; `parsedUrl` is never `null` / `undefined`. In fact, I'd rewrite all of this without an `if`, as just:

```js
parsedUrl.protocol = parsedUrl.protocol ?? 'https:';
```